addons/hr_holidays/models/hr_employee.py

```diff
@@ -204,3 +205,8 @@ def write(self, values):
             allocations = self.env['hr.leave.allocation'].sudo().search([('state', 'in', ['draft', 'confirm']), ('employee_id', 'in', self.ids)])
             allocations.write(hr_vals)
         return res
+
+    def toggle_active(self):
+        res = super(HrEmployeeBase, self).toggle_active()
+        self.hr_leave_ids.active = self.active
```

Comment:
It will not work when archiving/unarchiving multiple employees, instead you can call `toggle_active()` on `leave_ids` or do something like:
```py
active_employees = self.filtered('active')
active_employees.leave_ids.active = True
(self - active_employees).leave_ids.active = False
```